workers/data_refinery_workers/downloaders/array_express.py

```diff
@@ -21,34 +21,36 @@
 JOB_DIR_PREFIX = "downloader_job_"
 
 
-def _verify_batch_grouping(batches: List[Batch], job_id: int) -> None:
+def _verify_batch_grouping(batches: List[Batch], job: DownloaderJob) -> None:
     """All batches in the same job should have the same downloader url"""
     for batch in batches:
         if batch.download_url != batches[0].download_url:
-            logger.error(("A Batch doesn't have the same download URL as the other batches"
-                          " in downloader job #%d."),
-                         job_id)
-            raise ValueError("A batch doesn't have the same download url as other batches.")
+            failure_message = "A Batch doesn't have the same download URL as the other batches"
+            logger.error(failure_message + " in Downloader Job #%d.",
+                         job.id)
+            job.failure_reason = failure_message
+            raise ValueError(failure_message)
 
 
-def _download_file(download_url: str, file_path: str, job_id: int) -> None:
+def _download_file(download_url: str, file_path: str, job: DownloaderJob) -> None:
     try:
         logger.debug("Downloading file from %s to %s. (Job #%d)",
                      download_url,
                      file_path,
-                     job_id)
+                     job.id)
         target_file = open(file_path, "wb")
         with closing(urllib.request.urlopen(download_url)) as request:
             shutil.copyfileobj(request, target_file, CHUNK_SIZE)
     except Exception:
-        logging.exception("Exception caught while running Downloader Job #%d.",
-                          job_id)
+        logging.exception("Exception caught while downloading batch in Downloader Job #%d.",
+                          job.id)
+        job.failure_reason = "Exception caught while downloading batch"
```

Comment:
I think you misread my code a bit, because in lines 28-31 `job.failure_reason` does not include the `job.id`, only the message that gets logged does. In line 31 `job.failure_reason` gets set to `failure_message`, which has been defined as `"A Batch doesn't have the same download URL as the other batches"`. Within the call to `logger.error` I append `" in Downloader Job #%d."` to that base string and format it with the `job.id`.

Basically my thought is that the `job.failure_reason` does not need the id logged with it since it is on the job object and therefore has the ID. However I do want to log the `job.id` in the log message because otherwise I'll have no way to know which job failed.